File: 5-TOOLS-DEV/dev/health-ocr-parse/gpt.py
import sys
import re
from openai import OpenAI

if len(sys.argv) != 3:
    print("Usage: script.py data_file prompt_file")
    sys.exit(1)

data_filename = sys.argv[1]
prompt_filename = sys.argv[2]

with open(data_filename, 'r', encoding='utf-8') as f:
    data = f.read().strip()

with open(prompt_filename, 'r', encoding='utf-8') as f:
    prompt_content = f.read().strip()

# Split data into paragraphs separated by blank lines
paragraphs = re.split(r'\n\s*\n', data)

client = OpenAI()

outputs = []

for idx, paragraph in enumerate(paragraphs, start=1):
    print(f"Processing paragraph {idx} of {len(paragraphs)}")
    full_prompt = prompt_content + "\n" + paragraph

    messages = [
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": full_prompt}
    ]

    completion = client.chat.completions.create(
        model="gpt-4o-mini",
        messages=messages
    )

    output = completion.choices[0].message.content
    output_line = ' '.join(output.strip().split())
    outputs.append(output_line)

with open('output.txt', 'w', encoding='utf-8') as f:
    for output_line in outputs:
        f.write(output_line + '\n')

File: 5-TOOLS-DEV/dev/kml-event-database/step2/gpt.js
import fs from 'fs';
import path from 'path';
import OpenAI from 'openai';

// Initialize the OpenAI client
const openai = new OpenAI();

// Function to call OpenAI's ChatGPT API
async function callChatGPT(systemPrompt, userPrompt) {
  try {
    const completion = await openai.chat.completions.create({
      model: "gpt-4o-mini",  // Adjust the model name as needed
      messages: [
        { role: "system", content: systemPrompt },
        { role: "user", content: userPrompt },
      ],
    });

    // Return the response content from the API
    return completion.choices[0].message.content.trim();
  } catch (error) {
    console.error('Error calling ChatGPT API:', error.message);
    throw error;
  }
}

async function main() {
  // 1) Parse directory from command line
  if (process.argv.length < 3) {
    console.error("Usage: node script.js <relative_directory>");
    process.exit(1);
  }

  const inputDir = process.argv[2];

  // 2) Create output directory if it doesn't exist
  const outputDir = path.join(process.cwd(), 'output');
  if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir);
  }

  // 3) Read all files in the input directory
  let files;
  try {
    files = fs.readdirSync(inputDir);
  } catch (err) {
    console.error("Error reading directory:", err);
    process.exit(1);
  }

  // 4) Filter to only .txt files
  files = files.filter((file) => path.extname(file).toLowerCase() === '.txt');

  // Define our system prompt once
  const systemPrompt = "You are a helpful assistant";

  let index = 1;
  for (const file of files) {
    try {
      // 5) Read the file content to use as user prompt
      const filePath = path.join(inputDir, file);
      const userPrompt = fs.readFileSync(filePath, 'utf8');

      // 6) Call ChatGPT
      const response = await callChatGPT(systemPrompt, userPrompt);

      // 7) Write the response to the output directory, same filename
      const outputPath = path.join(outputDir, file);
      fs.writeFileSync(outputPath, response, 'utf8');

      // 8) Print progress
      console.log(`Processed file ${index}/${files.length}: ${file}`);
      index++;
    } catch (error) {
      console.error(`Error processing file '${file}': ${error.message}`);
    }
  }

  console.log('All files processed.');
}

// Invoke main
main().catch((error) => {
  console.error('Unexpected error:', error);
  process.exit(1);
});
